File: 0001-0500/0209. Minimum Size Subarray Sum.go
package main
// Medium
// Time:O(n), Space:O(1)
// https://leetcode.cn/problems/minimum-size-subarray-sum/

func minSubArrayLen(target int, nums []int) int {
    n := len(nums)
    minLen := n + 1
    sum := 0
    left := 0

    for right := 0; right < n; right++ {
        sum += nums[right]
        for sum >= target {
            minLen = min(minLen, right - left + 1)
            sum -= nums[left]
            left++
        }
    }
    if minLen == n + 1 {
        return 0
    }
    return minLen
}

File: 0001-0500/0209. Minimum Size Subarray Sum.py
# Time:O(n),Space:O(1)

from typing import List

class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        n = len(nums)
        left = 0
        cur_sum = 0
        ans = float('inf')

        for right in range(n):
            cur_sum += nums[right]

            while cur_sum >= target:
                ans = min(ans, right - left + 1)
                cur_sum -= nums[left]
                left += 1
        
        return ans if ans != float('inf') else 0
    